client/asset/eth/eth.go

```diff
@@ -700,9 +702,72 @@ func (eth *ExchangeWallet) SignMessage(coin asset.Coin, msg dex.Bytes) (pubkeys,
 
 // AuditContract retrieves information about a swap contract on the
 // blockchain. This would be used to verify the counter-party's contract
-// during a swap.
-func (*ExchangeWallet) AuditContract(coinID, contract, txData dex.Bytes, rebroadcast bool) (*asset.AuditInfo, error) {
-	return nil, asset.ErrNotImplemented
+// during a swap. coinID is expected to be the transaction id, and must
+// be the same as the hash of txData. contract is expected to be
+// (contractVersion|secretHash) where the secretHash uniquely keys the swap.
+func (eth *ExchangeWallet) AuditContract(coinID, contract, txData dex.Bytes, rebroadcast bool) (*asset.AuditInfo, error) {
+	tx := new(types.Transaction)
+	err := tx.UnmarshalBinary(txData)
+	if err != nil {
+		return nil, fmt.Errorf("AuditContract: failed to unmarshal transaction: %w", err)
+	}
+
+	txHash := tx.Hash()
+	if !bytes.Equal(coinID, txHash[:]) {
+		return nil, fmt.Errorf("AuditContract: coin id != txHash - coin id: %x, txHash: %x", coinID, tx.Hash())
+	}
+
+	initiations, err := dexeth.ParseInitiateData(tx.Data())
+	if err != nil {
+		return nil, fmt.Errorf("AuditContract: failed to parse initiate data: %w", err)
+	}
+
+	secretHash, _, err := decodeVersionedBytes(contract)
```

Comment:
Just a note here that a number of things come together because this `contract` input, as well as the  `asset.AuditInfo.Contract` where it is stored and pass back to the caller, contain the secret hash and contract version.
This is important because in #1320 the `asset.RedeemForm` input to `Redeem` gets this via `asset.Redemption.Spends` (this `*Auditinfo`).
As @buck54321 suggested, this does indeed appear to make `ToVersion` obsolete.  Will push a commit shortly, but since this PR actually implementing `AuditContract` isn't merged, it might not be obvious how it works when I push.

Reply:
With #1320 merged, should now use `dexeth.DecodeContractData` and remove the added helper